src/core/PixelStream.cpp

```diff
@@ -191,20 +207,34 @@ void PixelStream::adjustSegmentRendererCount(const size_t count)
     {
         segmentRenderers_.clear();
         for (size_t i=0; i<count; ++i)
-            segmentRenderers_.push_back( PixelStreamSegmentRendererPtr(new PixelStreamSegmentRenderer(renderContext_)) );
+            segmentRenderers_.push_back( boost::make_shared<PixelStreamSegmentRenderer>( ));
     }
 }
 
+void PixelStream::refreshSegmentsList( const deflect::PixelStreamSegments& segments )
+{
+    qDeleteAll( segmentsList_ );
```

Comment:
yes delete. Then I will resize the list instead and add an update() function to the Segement class, it should be a bit more efficient.